source/extensions/quic_listeners/quiche/udp_gso_batch_writer.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#pragma GCC diagnostic push
+// QUICHE allows unused parameters.
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+// QUICHE uses offsetof().
+#pragma GCC diagnostic ignored "-Winvalid-offsetof"
+// QUICHE allows ignored qualifiers
+#pragma GCC diagnostic ignored "-Wignored-qualifiers"
+
+// QUICHE doesn't mark override at QuicBatchWriterBase::SupportsReleaseTime()
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Winconsistent-missing-override"
+#elif defined(__GNUC__) && __GNUC__ >= 5
+#pragma GCC diagnostic ignored "-Wsuggest-override"
+#endif
+
+#include "quiche/quic/core/batch_writer/quic_gso_batch_writer.h"
+
+#ifdef __clang__
+#pragma clang diagnostic pop
+#endif
+
+#pragma GCC diagnostic pop
+
+#include "envoy/network/udp_packet_writer_handler.h"
+
+#include "common/protobuf/utility.h"
+#include "common/runtime/runtime_protos.h"
+
+namespace Envoy {
+namespace Quic {
+
+#define UDP_GSO_BATCH_WRITER_STATS(COUNTER, GAUGE, HISTOGRAM)                                      \
+  COUNTER(total_bytes_sent)                                                                        \
+  GAUGE(internal_buffer_size, NeverImport)                                                         \
+  HISTOGRAM(pkts_sent_per_batch, Unspecified)
```

Comment:
These should be documented somewhere? (This is related to my comment on better end user documentation.)

Reply:
I can add a comment over the UDP_GSO_BATCH_WRITER_STATS to document what each of the stats is recording. Is there any other place where you would like me to add the documentation for this?

Reply:
These need to be documented in the public RST documentation for this feature and/or QUIC. As I said elsewhere if this is not going to get done before you finish the week, I would prefer to just leave TODOs and mark all of the protos as not-implemented-hide so they don't show up in the docs.

Reply:
Added a comment here describing the details of the `UdpGsoBatchWriterStats`. Also added a TODO to move these comments into QUIC documentation when it is created in the future!